allennlp/modules/similarity_functions/bilinear.py

```diff
@@ -0,0 +1,62 @@
+from typing import Callable
+
+from overrides import overrides
+import torch
+from torch.nn.parameter import Parameter
+
+from allennlp.common import Params
+from allennlp.experiments import Registry
+from allennlp.modules import SimilarityFunction
+
+
+@Registry.register_similarity_function("bilinear")
+class Bilinear(SimilarityFunction):
```

Comment:
Possibly we should use `BilinearSimilarity` and `LinearSimilarity` for these because `torch.nn.Linear` and `torch.nn.Bilinear` already exist.